src/test/java/org/sagebionetworks/web/unitserver/StackConfigServiceTest.java

```diff
@@ -98,4 +110,32 @@ public void testGetStorageLocationSettingProperty()
       .get(SynapseClientImpl.DEFAULT_STORAGE_ID_PROPERTY_KEY);
     assertNotNull(defaultStorageId);
   }
+
+  @Test
+  public void testStackStatusCache() throws Exception {
+    verify(mockSynapse, times(0)).getCurrentStackStatus();
+
+    StackStatus status = stackConfigService.getCurrentStatus();
+    // call twice, and verify the backend is only called once due to cache
+    stackConfigService.getCurrentStatus();
+
+    assertEquals(mockStackStatus, status);
+    verify(mockSynapse, times(1)).getCurrentStackStatus();
+  }
+
+  @Test
+  public void testStackStatusErrorHandling() throws Exception {
```

Comment:
Verify error case (which is all I see from SWC locally due to the GEO restriction issue!)